File: Data Structures & Algorithms/search-2d-matrix/submission-0.java
class Solution {
    public boolean searchMatrix(int[][] matrix, int target) {
        int n = matrix[0].length;
        for(int mat[] : matrix){
            if(mat[0] <= target && target <= mat[n-1]){
                int left = 0;
                int right = n-1;
                while(left <= right){
                    int mid = left + (right - left) / 2;
                    if(mat[mid] == target) return true;
                    else if(mat[mid] < target) left = mid + 1;
                    else right = mid - 1;
                }
            }
        }
        return false;
    }
}

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-0.py
class Solution:
    def minSubArrayLen(self, target: int, nums: List[int]) -> int:
        count = 0
        res = float('inf')
        n = len(nums)
        left = 0

        for right in range(n):
            count += nums[right]

            while count >= target:
                count -= nums[left]
                res = min(res, right - left + 1)
                left += 1

            
        return res if res != float('inf') else 0